src/Symfony/Component/DomCrawler/Tests/CrawlerTest.php

```diff
@@ -815,6 +820,13 @@ public function testForm()
         } catch (\InvalidArgumentException $e) {
             $this->assertTrue(true, '->form() throws an \InvalidArgumentException if the node list is empty');
         }
+
+        try {
+            $crawler->filterXPath('//li/text()')->link();
+            $this->fail('->form() throws an \InvalidArgumentException if the selected node is not instance of \DOMElement');
+        } catch (\InvalidArgumentException $e) {
+            $this->assertTrue(true, '->form() throws an \InvalidArgumentException if the selected node is not instance of \DOMElement');
+        }
```

Comment:
Please create a separate test instead, and use ``setExpectedException`` (yeah, I know that the existing TestCase for the crawler is in a crappy state)